src/inmanta/ast/statements/call.py

```diff
@@ -238,6 +238,15 @@ def call_in_context(
             except UnknownException as e:
                 result.set_value(e.unknown, self.ast_node.location)
             except UnsetException as e:
+                # Don't handle it here!
+                # This exception is used by the scheduler to re-queue the unit
+                # If it is handled here, the re-queueing can not be done,
+                # leading to very subtle errors such as #2787
```

Comment:
Ideally we'd keep this just above the re-raising of the exception to make it clear what it refers to. Not terribly important though.